UWPCE-PythonCert-ClassRepos/SP_Online_PY210
# Anthony McKeever - Lesson 9 Content Submission

Final project done!  I had some fun doing this, and maybe got a little carried away with unit tests, but the code coverage is at 100% so I guess I can't complain.  :)

**_Squashed Commit History:_**

* Begin building out supporting mail room classes

* Continued work on MailRoom CLI

* Finish cli-mail room

* Test Donor in Donor Models

* Add init files

* More unit tests

Bug - Issue where tests will pass but cannot run app from CLI now.
	Probably due to how imports are being handled now that __init__
	files are here.

* Increase code coverage to 100%

* PEP8 alignment

Wouldn't you like to be a PEP8 too?

* Add force .coveragerc

* Minor refactorings

Allign with assignment demands and remove print statements from data
classes

* Flatten folder structure

Resolve import issues on CLI app.  Tests and app now work correctly.

* Final Refactor and documentation

Add documentation to all non-test methods
Move TestHelpers to test_donor_model since its mostly used there
Delete unnecessary test_helpers file
Refactor resource file acquisition.

* Add one last unit test for new method

Add test for FileHelpers.default_resource_file_path
	Although technically covered, we should explicitly test this
	logic.

* Appease the whitespace gods
